### PR TITLE
fcitx5-skk: init at 5.0.15

### LINKED ISSUE #167872: Packaging request: fcitx5-skk
**Project description**
fcitx5-skk is an input method engine for Fcitx5, which uses libskk as its backend.

It seems some of the fcitx5 engines (including fcitx5-skk) are missing. Probably the other engines also need to be added.

**Metadata**

* homepage URL:　https://github.com/fcitx/fcitx5-skk
* source URL:　https://github.com/fcitx/fcitx5-skk
* license: gpl3+
* platforms: linux

### PR DESCRIPTION
**Pull request by wattmto**
###### Description of changes

fcitx5-skk is an input method engine for Fcitx5, which uses libskk as its backend.

https://github.com/fcitx/fcitx5-skk

Fixes https://github.com/NixOS/nixpkgs/issues/167872.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by milran** [comment]
Ping @zer0-star and @Vonfry because you commented in the original issue.

**Comment by Vonfry** [comment]
You can add fcitx5-skk into https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/inputmethods/fcitx5/update.py to make it update by script with other packages togather.

File: pkgs/tools/inputmethods/fcitx5/fcitx5-skk.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv
+, fetchFromGitHub
+, pkg-config
+, cmake
+, extra-cmake-modules
+, gettext
+, fcitx5
+, fcitx5-qt
+, libskk
+, qtbase
```
Comment (Vonfry) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-skk.nix:10]: > Qt5 (optional), for rule and dictionary configuration UI.
According to the documentation of fcitx5-skk, qtbase is optional with cmake flag `ENABLE_QT`. 

By the way, should be qt6 added as well?

**Review by Vonfry — CHANGES_REQUESTED** [review]
It works for me.

**Review by Vonfry — APPROVED** [review]

File: pkgs/tools/inputmethods/fcitx5/fcitx5-skk.nix (hunk 1 above)
Comment (milran) [reply pkgs/tools/inputmethods/fcitx5/fcitx5-skk.nix:10]: Fixed.

It seems we have to add `fcitx5-qt` in https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/qt6-packages.nix before build `fcitx5-skk` with Qt6.

**Comment by milran** [comment]
@Vonfry sorry, I made a minor format fix. Could you check this again?

**Review by Vonfry — APPROVED** [review]

File: pkgs/tools/inputmethods/fcitx5/fcitx5-skk.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  buildInputs = [
+    fcitx5
+    libskk
+  ] ++ lib.optionals enableQt [
+    fcitx5-qt
+    qtbase
+  ];
+
+  cmakeFlags = [
+    "-DENABLE_QT=${ toString enableQt }"
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/fcitx5/fcitx5-skk.nix:43]:
```suggestion
    "-DENABLE_QT=${toString enableQt}"
```

**Comment by milran** [comment]
@SuperSandro2000 Thank you! fixed.

**Pull request merged**